gtest/src/log.rs

```diff
@@ -396,6 +396,40 @@ impl RunResult {
             .flat_map(DecodedCoreLog::try_from_log)
             .collect()
     }
+
+    /// Trying to get the panic log.
+    fn panic_log(&self) -> Option<&CoreLog> {
+        let [core_log] = &self.log[..] else {
+            return None;
+        };
+        let is_panic = matches!(
+            core_log.reply_code(),
+            Some(ReplyCode::Error(ErrorReplyReason::Execution(
+                SimpleExecutionError::UserspacePanic
+            )))
+        );
+        is_panic.then_some(core_log)
+    }
+
+    /// If the main message panicked.
+    pub fn panicked(&self) -> bool {
+        self.panic_log().is_some()
+    }
+
+    /// Asserts that the main message panicked with a given message.
+    #[track_caller]
+    pub fn assert_panicked_with(&self, msg: impl Into<String>) {
+        if let Some(log) = self.panic_log() {
+            let msg = msg.into();
+            let payload =
+                String::from_utf8(log.payload().into()).expect("Unable to decode panic message");
+
+            assert!(
+                payload.contains(&format!("panicked with '{msg}'")),
```

Comment:
starts with

Reply:
Is there not something before this, like "Thread ... panicked with"?

Reply:
"Panic occurred: panicked with '{message}'[ at '{loc}']"
Just for notice, I don't see any reason to assert paths